### ISSUE TITLE
mtree: invalid keyword (for sha256)

### ISSUE THREAD

**Issue by vbatts** [systemd/casync#101]
from `casync mtree myhost.caidx`:
one of the `type=file` entries has a `sha512256digest=b6d59d2d0a0d29fa315bcfcc05e7885e1d29acab6b72d5ce01777364c3aa0980`

That should be `sha256digest=`. And while some BSD versions don't support it yet, go-mtree does support a `sha512digest=` keyword.

**Comment by poettering** [comment]
ah, uh, so I changed the default hash function casync uses from SHA256 to SHA512/256, but BSD mtree don't know such a property. I didn't really know what to do on this, so I invented my own field, and called it "sha512256digest" following the naming logic of the other fields (but dropping the slash).

Yes, I know it's uncool to extend the mtree format like this, and without any noise about it. but i think i did the obvious choice here...

It would be great if go-mtree would support a similar field, even if it's not available in the "offical" BSD mtree.

SHA512/256 is a standardized part of the SHA-2 set of hash functions. It's a truncated version of SHA512 that uses a different initialization vector. It's substantially faster to calculate than SHA256 on modern 64bit processors, though it otherwise has very similar properties. Given that casync crunches a lot of data, replacing SHA256 by SHA512/256 (we support both however) made a lot of sense.
